cluster-autoscaler/cloudprovider/azure/test/templates/cluster-template-prow-aks-aso-cluster-autoscaler.yaml

```diff
@@ -31,6 +30,7 @@ spec:
       dnsPrefix: ${CLUSTER_NAME}
       identity:
         type: SystemAssigned
+      kubernetesVersion: "${KUBERNETES_VERSION}"
```

Comment:
Moving the Kubernetes version here allows us to use AKS `MAJOR.MINOR` versions where CAPI requires `spec.version` to be formatted as `MAJOR.MINOR.PATCH`. This allows us to automatically inherit new patch releases when they become available in AKS.